tests: arm_sw_vector_relay: Take into account CONFIG_SRAM_VECTOR_TABLE

ydamigos opened the pull request:
Test arm_sw_vector_relay started to fail on our internal CI when we selected CONFIG_SRAM_VECTOR_TABLE as a default option for our (unreleased yet) SoC.

Test needs to take into account vector table RAM relocation.
New test case was added and tested on `qemu_cortex_m3` platform.

ydamigos committed:
tests: arm_sw_vector_relay: Take into account CONFIG_SRAM_VECTOR_TABLE

Take into account vector table RAM relocation.

Signed-off-by: Ioannis Damigos <ioannis.damigos.uj@renesas.com>

wearyzen reviewed (commented):
Hi @ydamigos, thank you for suggesting the change.
Could you please add some more context in the commit message or pr description to help understand what testcase (or which board) led to this change? If there is a github issue related to this then it would be good to reference it here and if not then how was the issue discovered. It would also be good to add a testcase in tests/arch/arm/arm_sw_vector_relay/testcase.yaml with your board to cover this scenario.

ydamigos committed:
tests: arm_sw_vector_relay: Add test case for RAM vector table relocation

Add test case to cover RAM vector table relocation when
CONFIG_SRAM_VECTOR_TABLE is selected

Signed-off-by: Ioannis Damigos <ioannis.damigos.uj@renesas.com>

ydamigos commented:
> Hi @ydamigos, thank you for suggesting the change. Could you please add some more context in the commit message or pr description to help understand what testcase (or which board) led to this change? If there is a github issue related to this then it would be good to reference it here and if not then how was the issue discovered. It would also be good to add a testcase in tests/arch/arm/arm_sw_vector_relay/testcase.yaml with your board to cover this scenario.

@wearyzen Test arm_sw_vector_relay started to fail on our internal CI when we selected CONFIG_SRAM_VECTOR_TABLE as a default option for our (unreleased yet) SoC.

I added a test case and tested it on `qemu_cortex_m3` platform.
